language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -386,11 +388,17 @@ genDefDataType curPkgId conName mod tpls def =
                             ] ++
                             ["};"]
                         associatedTypes =
-                          maybe [] (\key ->
-                              [ "export namespace " <> conName <> " {"
-                              , "  export type Key = " <> fst (genType (moduleName mod) (tplKeyType key)) <> ""
-                              , "}"
-                              ]) (tplKey tpl)
+                          let tT = conName
+                              tK = tT <> ".Key"
```

Comment:
I assume this will just be undefined if we do not have a key? Is this tested?

Reply:
In the event there is no key, then the definition in play is `export type Key = unknown` which is consistent with the definitions of `Event<>` and `CreateEvent<>`. Yes, tested.